Feat/webhooks verification (#1048)
codatio/codat-docs: docs/using-the-api/webhooks/create-consumer.md

## Webhook signature verification

A webhook signature is your way to verify that the messages are sent by Codat and helps you avoid impersonation or replay attacks. We sign every webhook and its metadata with a unique security key for each endpoint and include timestamps for when the message attempt occurred.

> You can use this signature to verify that the message truly came from Codat and process it. To do this, we advise using a library called Svix to verify the event was sent by Codat.

Reviewer: ```suggestion
You can use this signature to verify that the message truly came from Codat before processing it. To do the verification, we suggest using a library called Svix.
```

### Install library

<Tabs>

<TabItem value="nodejs" label="JavaScript">

##### NPM
```sh
npm install svix
```

##### Yarn
```sh
yarn add svix
```
</TabItem>

<TabItem value="python" label="Python">

```sh
pip install svix
```
</TabItem>

<TabItem value="csharp" label="C#">

```sh
dotnet add package Svix
```
</TabItem>

<TabItem value="go" label="Go">

```sh
go get github.com/svix/svix-webhooks/go
```
</TabItem>

<TabItem value="java" label="Java">

#### Gradle

Add this dependency to your project's build file
```sh
implementation "com.svix:svix:0.x.y"
```

#### Maven

> Add this dependency to your project's POM

Reviewer: ```suggestion
Add this dependency to your project's POM:
```

```xml
<dependency>
  <groupId>com.svix</groupId>
  <artifactId>svix</artifactId>
  <version>0.x.y</version>
</dependency>
```
</TabItem>

<TabItem value="rust" label="Rust">

```sh
svix = "0"
```
</TabItem>

<TabItem value="kotlin" label="Kotlin">

#### Gradle

> Add this dependency to your project's build file

Reviewer: ```suggestion
Add this dependency to your project's build file:
```

```sh
implementation "com.svix.kotlin:svix-kotlin:0.x.y"
```

#### Maven

> Add this dependency to your project's POM

Reviewer: ```suggestion
Add this dependency to your project's POM:
```

```xml
<dependency>
  <groupId>com.svix.kotlin</groupId>
  <artifactId>svix-kotlin</artifactId>
  <version>0.x.y</version>
</dependency>
```
</TabItem>

<TabItem value="ruby" label="Ruby">

```sh
gem install svix
```
</TabItem>

<TabItem value="php" label="PHP">

```sh
composer require svix/svix
```
</TabItem>

<TabItem value="cli" label="CLI">

#### Homebrew

```sh
brew install svix/svix/svix
```

#### Scoop

```sh
scoop bucket add svix https://github.com/svix/scoop-svix.git
scoop install svix
```

</TabItem>

</Tabs>

### Verify webhook

> To verify incoming webhooks first retrieve the secret key for your endpoint. In the [Codat Portal](https://app.codat.io/monitor/events) navigate to **Monitor > Webhooks > Events**, select the endpoint you want to verify and copy the **Signing secret** for use in your application.

Reviewer: ```suggestion
To verify incoming webhooks, retrieve the secret key for your endpoint first. 

In the [Codat Portal](https://app.codat.io/monitor/events), navigate to **Monitor > Webhooks > Events**, click the endpoint you want to verify, and copy the **Signing secret** from the endpoint's detailed view.
```

 
  ![A fragment of the UI that highlights where to copy the signing secret](/img/use-the-api/0054-endpoint-detail-signing-secret.png)

> The webhook is verified by passing the secret key, request body and headers to the verification library as demonstrated below.  

Reviewer: ```suggestion
Next, you need to pass the secret key, request body, and headers to the verification library as demonstrated below.  
```

:::info Use the raw request body

> You need to use the raw request body when verifying webhooks, as the cryptographic signature is sensitive to even the slightest changes. You should watch out for frameworks that parse the request as JSON and then stringify it because this too will break the signature verification.

Reviewer: ```suggestion
You need to use the raw request body when verifying webhooks because the cryptographic signature is very sensitive to changes. 

Watch out for frameworks that parse the request as JSON and then stringify it, because this will also break the signature verification.
```

A webhook signature is your way to verify that the messages are sent by Codat and helps you avoid impersonation or replay attacks. We sign every webhook and its metadata with a unique key for each endpoint and include timestamps for when the message attempt occurred.

> You can use this signature to verify that the message truly came from Codat and process it. To do this, we advise using Svix's library to verify the event was sent by Codat. 

Reviewer 2: This is the first mention of Svix on this page, introducing Svix's lib here without any context.

If you don't want to explain who Svix is then the below language may work. You'd also need to change line 75

```suggestion
You can use this signature to verify that the message truly came from Codat and process it. To do this, we advise using a library called Svix to verify the event was sent by Codat. 
```

```
</TabItem>

<TabItem value="java" label="Java">

#### Gradle

Add this dependency to your project's build file
```sh
implementation "com.svix:svix:0.x.y"
```

#### Maven

Add this dependency to your project's POM
```xml
<dependency>
  <groupId>com.svix</groupId>
  <artifactId>svix</artifactId>
  <version>0.x.y</version>
</dependency>
```
</TabItem>

<TabItem value="rust" label="Rust">

```sh
svix = "0"
```
</TabItem>

<TabItem value="kotlin" label="Kotlin">

#### Gradle

Add this dependency to your project's build file
```sh
implementation "com.svix.kotlin:svix-kotlin:0.x.y"
```

#### Maven

Add this dependency to your project's POM
```xml
<dependency>
  <groupId>com.svix.kotlin</groupId>
  <artifactId>svix-kotlin</artifactId>
  <version>0.x.y</version>
</dependency>
```
</TabItem>

<TabItem value="ruby" label="Ruby">

```sh
gem install svix
```
</TabItem>

<TabItem value="php" label="PHP">

```sh
composer require svix/svix
```
</TabItem>

<TabItem value="cli" label="CLI">

#### Homebrew

```sh
brew install svix/svix/svix
```

#### Scoop

```sh
scoop bucket add svix https://github.com/svix/scoop-svix.git
scoop install svix
```

</TabItem>

</Tabs>

> ### Verify webhook

Reviewer 2: Rather than just instruct on how to do this, it's probably useful to contextualise it with why you might want to do it.

Each endpoint uses its own secret to verify the webhook. In the [Codat Portal](https://app.codat.io/monitor/events) navigate to **Monitor > Webhooks > Events**, select the endpoint you want to verify and copy the **Signing secret** for use in your application.

> Verify Codat's webhooks by passing the signing secret to the webhook constructor and use the verify method to confirm the event was sent by Codat.

Reviewer 2: What does this actually mean?
